presto-hive/src/main/java/com/facebook/presto/twitter/hive/util/TTransportPool.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive.util;
+
+import com.google.common.net.HostAndPort;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.PooledObjectFactory;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.thrift.transport.TTransport;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Utility class to handle creating and caching the UserGroupInformation object.
+ */
+public class TTransportPool
+{
+    private final ConcurrentMap<String, ObjectPool<TTransport>> pools = new ConcurrentHashMap();
```

Comment:
You haven't exposed the config setting for the pool size and other stuff right? So right now ObjectPool<TTransport> can grow without bound.